huaweicloud/data_source_huaweicloud_obs_bucket_object.go

```diff
@@ -0,0 +1,144 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/obs"
+)
+
+func dataSourceObsBucketObject() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceObsBucketObjectRead,
+
+		Schema: map[string]*schema.Schema{
+			"body": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"key": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"storage_class": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"acl": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"content_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Attribute parameters are not returned in one interface.
+// Three interfaces need to be called to get all parameters
+func dataSourceObsBucketObjectRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	obsClient, err := config.newObjectStorageClient(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud OBS client: %s", err)
+	}
+	bucket := d.Get("bucket").(string)
+	key := d.Get("key").(string)
+	uniqueId := bucket + "/" + key
+
+	resp, err := obsClient.GetObject(&obs.GetObjectInput{
+		GetObjectMetadataInput: obs.GetObjectMetadataInput{
+			Bucket: bucket,
+			Key:    key,
+		},
+	})
+	if err != nil {
+		return getObsError("Error get object info of OBS bucket", bucket, err)
+	}
+	log.Printf("[DEBUG] Data Source Reading OBS Bucket Object : %#v", resp)
+
+	d.Set("content_type", resp.ContentType)
+	if resp.Body != nil {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return getObsError("Error get body of OBS bucket object", bucket, err)
+		}
+		d.Set("body", string(body))
+	}
```

Comment:
The content of an object maybe very large, please do some test on a huge object, thanks